partner_time_to_pay/models/res_partner.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    d2p_life = fields.Float(
+        compute='_compute_d2x',
+        string='AVG Days to Payable (lifetime)'
+    )
+    d2p_ytd = fields.Float(
+        compute='_compute_d2x',
+        string='AVG Days to Payable (YTD)'
+    )
+    d2r_life = fields.Float(
+        compute='_compute_d2x',
+        string='AVG Days to Receivable (lifetime)'
+    )
+    d2r_ytd = fields.Float(
+        compute='_compute_d2x',
+        string='AVG Days to Receivable (YTD)'
+    )
+
+    def _compute_d2x(self):
+        for partner in self:
+            partner.d2p_ytd, partner.d2p_life = \
+                self._compute_d2x_per_invoice_type(partner, 'out_invoice')
```

Comment:
You can do both calculations in 1 function by combining the calculation of 'out_invoice' and 'in_invoice' in this function and then returning all 4 variables at once. Then you wouldn't need 2 searches on account_invoice tables.

Reply:
@PieterPaulussen I think it's better to differentiation our logic in sub-methods.

Reply:
@PieterPaulussen You are the OCA expert here ;-) What do you suggest?

Reply:
Well, as I stated above, I believe the functions should be merged. 
PS: I would hardly state that I'm the OCA expert though. ;-)